content/blog/entries/science-museum-implementation/contents.lr

```diff
@@ -0,0 +1,115 @@
+or partition using the parameters, etc. Since their API parameter has ```page[number]``` paging would be an appropriate choice with  max size 
+as 100 it would require around 800 pages to get all the objects but then certain glitches with their API doesn't allow us to go after 
+page 50.This would mean we would get only 5000 objects and around 17000 images. 
+
+[API page-50](https://collection.sciencemuseumgroup.org.uk/search/image_license?page[size]=100&page[number]=50)
+
+[API page-51](https://collection.sciencemuseumgroup.org.uk/search/image_license?page[size]=100&page[number]=51)
+
+So we need to find the way to divide the collection into subsets such that each subset has less than or equal to 5000 objects.
+Luckily, the API had another set of parameters ```date[from]``` and ```date[to]``` which represents the time period of the object. 
+Querying the API through different time period at the same time ensuring that records in that time period don't exceed 5000 solves the problem, starting
+from year 0 to year 2020 by trial and error method suitable year range was chosen.
+
+```
+                                            YEAR_RANGE = [
+                                                (0, 1500),
+                                                (1500, 1750),
+                                                (1750, 1825),
+                                                (1825, 1850),
+                                                (1850, 1875),
+                                                (1875, 1900),
+                                                (1900, 1915),
+                                                (1915, 1940),
+                                                (1940, 1965),
+                                                (1965, 1990),
+                                                (1990, 2020)
+                                            ]
+```
+
+With this we confirm the implementation method, but before writing the script we need to know what are the different license provided 
+by them and check if it matches with our licenses and version, since they haven't mentioned about their licenses and version, running a
+test script to get all the licenses and grouping them would do the work.
+
+```
++-----------------+----------+
+| license_version | count(1) |
++-----------------+----------+
+| CC-BY-NC-ND 2.0 |      210 |
+| CC-BY-NC-ND 4.0 |     2376 |
+| CC-BY-NC-SA 2.0 |        1 |
+| CC-BY-NC-SA 4.0 |    61694 |
++-----------------+----------+
+```
+
+Since the licenses and their versions are confirmed, we can start the implementation.
+
+## Implementation
+The implementation is quite simple in nature, we loop the through the ```YEAR_RANGE``` and get all the records for that period and 
+pass it on to an object data handler method that extracts the necessary details from the record and store it in the ```ImageStore```
```

Comment:
Please introduce the `ImageStore` concept (or link to the relevant part of the repo).